### ISSUE TITLE
Upgrade 4.5 -> 4.6 UPI vSphere fails, master upgrade is stuck

### ISSUE THREAD

**Issue by evertmulder** [openshift/okd#412]
**Describe the bug**
I have encountered several issues during the upgrade from 4.5 -> 4.6 including the missing hostname (changed to fedora) and rpm-tree issue due to repo's not being disabled. After fixing the hostnames and disabling the repo's all workers were upgraded, but the first master is still not progressing.

```
oc get clusterversion
NAME      VERSION                         AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.5.0-0.okd-2020-10-15-235428   True        True          2d9h    Unable to apply 4.6.0-0.okd-2020-11-27-200126: an unknown error has occurred: MultipleErrors
```

MCO daemon from master1 is reporting this error (master 1 was manual enabled for scheduling at one point trying to get the upgrade going)
```
oc logs -f --tail 10 -n openshift-machine-config-operator -c machine-config-daemon machine-config-daemon-q4d22
I1203 21:42:15.264110   11076 daemon.go:507] Detected a login session before the daemon took over on first boot
I1203 21:42:15.264126   11076 daemon.go:508] Applying annotation: machineconfiguration.openshift.io/ssh
I1203 21:42:15.272852   11076 daemon.go:1085] Validating against current config rendered-master-95343cca24b2f3b1bd4e79604349dbec
E1203 21:42:15.272890   11076 writer.go:135] Marking Degraded due to: unexpected on-disk state validating against rendered-master-95343cca24b2f3b1bd4e79604349dbec: expected target osImageURL "quay.io/openshift/okd-content@sha256:8cf7e06dd4095f2cd54e13fdb6fd313abbeb6e03d568f17956d97433623093c2", have "quay.io/openshift/okd-content@sha256:880f0e1cfb4a5c8c22dd9cbb54c2294bcbbcfef30b126d5c04d3160f84438c6a"
``` 

```
oc get nodes -o wide
NAME                           STATUS                     ROLES    AGE    VERSION                     INTERNAL-IP     EXTERNAL-IP     OS-IMAGE                         KERNEL-VERSION           CONTAINER-RUNTIME
master1.okd.otaphoenix.local   Ready,SchedulingDisabled   master   111d   v1.19.0-rc.2+9f84db3-1075   10.161.111.54   10.161.111.54   Fedora CoreOS 33.20201124.10.1   5.9.9-200.fc33.x86_64    cri-o://1.19.0
master2.okd.otaphoenix.local   Ready                      master   111d   v1.18.3                     10.161.111.55   10.161.111.55   Fedora CoreOS 32.20200629.3.0    5.6.19-300.fc32.x86_64   cri-o://1.18.2
master3.okd.otaphoenix.local   Ready                      master   111d   v1.18.3                     10.161.111.56   10.161.111.56   Fedora CoreOS 32.20200629.3.0    5.6.19-300.fc32.x86_64   cri-o://1.18.2
worker1.okd.otaphoenix.local   Ready                      worker   111d   v1.19.0-rc.2+9f84db3-1075   10.161.111.57   10.161.111.57   Fedora CoreOS 33.20201124.10.1   5.9.9-200.fc33.x86_64    cri-o://1.19.0
worker2.okd.otaphoenix.local   Ready                      worker   111d   v1.19.0-rc.2+9f84db3-1075   10.161.111.58   10.161.111.58   Fedora CoreOS 33.20201124.10.1   5.9.9-200.fc33.x86_64    cri-o://1.19.0
worker3.okd.otaphoenix.local   Ready                      worker   111d   v1.19.0-rc.2+9f84db3-1075   10.161.111.59   10.161.111.59   Fedora CoreOS 33.20201124.10.1   5.9.9-200.fc33.x86_64    cri-o://1.19.0
worker4.okd.otaphoenix.local   Ready                      worker   111d   v1.19.0-rc.2+9f84db3-1075   10.161.111.60   10.161.111.60   Fedora CoreOS 33.20201124.10.1   5.9.9-200.fc33.x86_64    cri-o://1.19.0
```

**Version**
4.5.0-0.okd-2020-10-15-235428   -->  4.6.0-0.okd-2020-11-27-200126

**How reproducible**
n.a.

**Log bundle**
```
oc adm must-gather --image-stream=openshift/must-gather
error: unable to resolve image stream 'openshift/must-gather': the server is currently unable to handle the request (get imagestreams.image.openshift.io must-gather)
```
UPDATE 8 dec 2020: The must-gather is working now: https://drive.google.com/file/d/1mc5WSqTtm9ob2wlASsk-GCN6ejLo7ktV/view?usp=sharing

Any hint to get the masters running again?

**Comment by kai-uwe-rommel** [comment]
On one of my upgrades of this kind (4.5 to 4.6 vSphere UPI) also the first master was stuck.
I simply rebooted it and then it progressed and completed the upgrade normally.

**Comment by evertmulder** [comment]
> I simply rebooted it and then it progressed and completed the upgrade normally.

A reboot did not solve the issue for me ...

**Comment by evertmulder** [comment]
Hmm still no idea on how to get my cluster running again. Would it be possible to just remove the master1 completely and reinstall a new master4 to create a  working cluster again?

**Comment by fragbyte** [comment]
> Hmm still no idea on how to get my cluster running again. Would it be possible to just remove the master1 completely and reinstall a new master4 to create a working cluster again?

We got the upgrade progressing by SSH-ing to the nodes and changing their hostnames to their node names. Apparently during the upgrade the new version starts out with the hostname "fedora". We have no DHCP running for the cluster.

**Comment by evertmulder** [comment]
> We got the upgrade progressing by SSH-ing to the nodes and changing their hostnames to their node names. Apparently during the upgrade the new version starts out with the hostname "fedora". We have no DHCP running for the cluster.

Yes I already did this for all nodes, and this worked for the worker nodes, but the master node is still not progressing...

**Comment by evertmulder** [comment]
Reading a tread on slack from Elytscha Smith, describing having a similar error I tried the following:
```
export HTTP_PROXY=my.proxy.ip
export HTTPS_PROXY=my.proxy.ip
/run/bin/machine-config-daemon pivot quay.io/openshift/okd-content@sha256:8cf7e06dd4095f2cd54e13fdb6fd313abbeb6e03d568f17956d97433623093c2
reboot
```

 When following the MCO daemon log now seems to be stuck on another point:
```
...
E1208 07:29:35.260788    2450 writer.go:135] Marking Degraded due to: getting pending state from journal: invalid character '\x1b' looking for beginning of value
E1208 07:30:35.276195    2450 writer.go:135] Marking Degraded due to: getting pending state from journal: invalid character '\x1b' looking for beginning of value
E1208 07:31:35.292079    2450 writer.go:135] Marking Degraded due to: getting pending state from journal: invalid character '\x1b' looking for beginning of value
...
```

**Comment by vrutkovs** [comment]
Looks similar to https://bugzilla.redhat.com/show_bug.cgi?id=1886207. Try removing the truncated journal with ` journalctl --vacuum..` and verify with `journalctl --verify`

**Comment by evertmulder** [comment]
Woot! this worked.
```
journalctl --vacuum-time 1h
journalctl --verify
# this finished without issues now
```

The MCO the was giving the following error:
```
I1208 15:11:46.934553    2450 daemon.go:1085] Validating against current config rendered-master-95343cca24b2f3b1bd4e79604349dbec
E1208 15:11:47.313755    2450 daemon.go:1505] content mismatch for file "/etc/systemd/system/kubelet.service.d/10-mco-default-env.conf" (-want +got):
  bytes.Join({
        "[Unit]\n",
-       "Description=MCO environment configuration\n",
        "[Service]\nEnvironment=HTTP_PROXY=http://10.151.87.161:8080\nEnvir",
        "onment=HTTPS_PROXY=http://10.151.87.161:8080\nEnvironment=NO_PROX",
        ... // 291 identical bytes
  }, "")
E1208 15:11:47.313792    2450 writer.go:135] Marking Degraded due to: unexpected on-disk state validating against rendered-master-95343cca24b2f3b1bd4e79604349dbec: content mismatch for file "/etc/systemd/system/kubelet.service.d/10-mco-default-env.conf"
```
After adding `Description=MCO environment configuration` to `/etc/systemd/system/kubelet.service.d/10-mco-default-env.conf` the MCO did finish.

**Comment by vrutkovs** [comment]
>After adding Description=MCO environment configuration 

A bit odd - it should have happened automatically. However this is required only once during 4.5 -> 4.6 upgrade, so I think we can fix that later.

Do you mind if I close this?

**Comment by evertmulder** [comment]
The MCO seems fine now, but not all operators did upgrade successful:
```
oc get co
NAME                                       VERSION                         AVAILABLE   PROGRESSING   DEGRADED   SINCE
authentication                             4.6.0-0.okd-2020-11-27-200126   False       True          True       5d2h
cloud-credential                           4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
cluster-autoscaler                         4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
config-operator                            4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
console                                    4.6.0-0.okd-2020-11-27-200126   False       False         True       6h52m
csi-snapshot-controller                    4.6.0-0.okd-2020-11-27-200126   True        False         False      4h28m
dns                                        4.6.0-0.okd-2020-11-27-200126   True        False         False      7d6h
etcd                                       4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
image-registry                             4.6.0-0.okd-2020-11-27-200126   True        False         True       9h
ingress                                    4.6.0-0.okd-2020-11-27-200126   True        False         False      177m
insights                                   4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
kube-apiserver                             4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
kube-controller-manager                    4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
kube-scheduler                             4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
kube-storage-version-migrator              4.6.0-0.okd-2020-11-27-200126   True        False         False      178m
machine-api                                4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
machine-approver                           4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
machine-config                             4.6.0-0.okd-2020-11-27-200126   True        False         False      4h20m
marketplace                                4.6.0-0.okd-2020-11-27-200126   True        False         False      7d6h
monitoring                                 4.6.0-0.okd-2020-11-27-200126   False       False         True       4h28m
network                                    4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
node-tuning                                4.6.0-0.okd-2020-11-27-200126   True        False         False      7d6h
openshift-apiserver                        4.6.0-0.okd-2020-11-27-200126   False       False         False      4h24m
openshift-controller-manager               4.6.0-0.okd-2020-11-27-200126   True        False         False      7d6h
openshift-samples                          4.6.0-0.okd-2020-11-27-200126   True        False         False      7d6h
operator-lifecycle-manager                 4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
operator-lifecycle-manager-catalog         4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
operator-lifecycle-manager-packageserver   4.6.0-0.okd-2020-11-27-200126   True        False         False      8s
service-ca                                 4.6.0-0.okd-2020-11-27-200126   True        False         False      116d
storage                                    4.6.0-0.okd-2020-11-27-200126   True        False         False      7d6h
```

Perhaps some open an new issue for this?
